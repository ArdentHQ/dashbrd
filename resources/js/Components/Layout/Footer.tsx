import cn from "classnames";
import { useTranslation } from "react-i18next";
import { Icon } from "@/Components/Icon";
import { Link } from "@/Components/Link";
import { PoweredByIcons } from "@/Components/PoweredByIcons";
import { appName } from "@/Utils/app";

interface FooterProperties {
    withActionToolbar?: boolean;
}

const FooterSeparator = ({ className }: { className?: string }): JSX.Element => (
    <span
        className={cn(
            "mx-2 h-1.25 w-1.25 rounded-full border border-theme-secondary-300 bg-theme-secondary-300 sm:mx-4",
            className,
        )}
    ></span>
);

export const Footer = ({ withActionToolbar = false }: FooterProperties): JSX.Element => {
    const { t } = useTranslation();

    return (
        <footer
            data-testid="Footer"
            className={cn("w-full bg-theme-secondary-50", {
                "pb-16": withActionToolbar,
            })}
        >
            <div className="mx-auto w-full max-w-content items-end justify-between px-6 py-4 text-sm sm:px-8 md-lg:flex md-lg:items-center lg:px-8 2xl:px-0">
                <div className="text-theme-secondary-700">
                    {new Date().getFullYear()} Â©{" "}
                    <Link
                        variant="link"
                        href={t("urls.landing")}
                        textColor="text-theme-hint-600"
                    >
                        {appName()}
                    </Link>
                    . {t("footer.all_rights_reserved")}.
                </div>

                <div className="mt-2 flex-1 items-center justify-between sm:flex md:mt-0">
                    <div className="flex items-center">
                        <FooterSeparator className="hidden md-lg:block" />
                        <Link
                            variant="link"
                            href={t("urls.privacy_policy")}
                            useAnchorTag
                        >
                            {t("footer.privacy_policy")}
                        </Link>

                        <FooterSeparator />

                        <Link
                            variant="link"
                            href={t("urls.terms_of_service")}
                            useAnchorTag
                        >
                            {t("footer.terms_of_service")}
                        </Link>

                        <div className="hidden items-center sm:flex">
                            <FooterSeparator />

                            <span className="pr-2 text-sm font-medium leading-5.5 text-theme-secondary-700">
                                {t("footer.powered_by")}
                            </span>

                            <PoweredByIcons />
                        </div>
                    </div>

                    <div className="mt-2 flex flex-col justify-between xs:h-5 xs:flex-row xs:items-center sm:mt-0 sm:justify-end lg:justify-start">
                        <div className="flex sm:hidden">
                            <span className="pr-2 text-sm font-medium leading-5.5 text-theme-secondary-700">
                                {t("footer.powered_by")}
                            </span>
                            <PoweredByIcons />
                        </div>
                        <div className="mt-2 flex space-x-4 xs:mt-0 md:space-x-0">
                            <a
                                href={t("urls.twitter").toString()}
                                className="transition-default text-theme-secondary-700 hover:text-theme-hint-700 md:hidden"
                                target="_blank"
                                rel="noopener nofollow noreferrer"
                            >
                                <span className="sr-only">Twitter</span>
                                <Icon
                                    name="Twitter"
                                    aria-hidden="true"
                                />
                            </a>

                            <a
                                href={t("urls.twitter").toString()}
                                className="button-icon group hidden border-transparent bg-transparent text-theme-secondary-700 hover:text-theme-hint-900 md:flex"
                                target="_blank"
                                rel="noopener nofollow noreferrer"
                            >
                                <span className="sr-only">Twitter</span>
                                <Icon
                                    name="Twitter"
                                    aria-hidden="true"
                                />
                            </a>

                            <a
                                href={t("urls.github").toString()}
                                className="transition-default text-theme-secondary-700 hover:text-theme-hint-700 md:hidden"
                                target="_blank"
                                rel="noopener nofollow noreferrer"
                            >
                                <span className="sr-only">GitHub</span>
                                <Icon
                                    name="GitHub"
                                    aria-hidden="true"
                                />
                            </a>

                            <a
                                href={t("urls.github").toString()}
                                className="button-icon group !ml-0 hidden border-transparent bg-transparent text-theme-secondary-700 hover:text-theme-hint-900 md:flex"
                                target="_blank"
                                rel="noopener nofollow noreferrer"
                            >
                                <span className="sr-only">GitHub</span>
                                <Icon
                                    name="GitHub"
                                    aria-hidden="true"
                                />
                            </a>

                            <a
                                href={t("urls.discord").toString()}
                                className="transition-default text-theme-secondary-700 hover:text-theme-hint-700 md:hidden"
                                target="_blank"
                                rel="noopener nofollow noreferrer"
                            >
                                <span className="sr-only">Discord</span>
                                <Icon
                                    name="Discord"
                                    aria-hidden="true"
                                />
                            </a>

                            <a
                                href={t("urls.discord").toString()}
                                className="button-icon group !ml-0 hidden border-transparent bg-transparent text-theme-secondary-700 hover:text-theme-hint-900 md:flex"
                                target="_blank"
                                rel="noopener nofollow noreferrer"
                            >
                                <span className="sr-only">Discord</span>
                                <Icon
                                    name="Discord"
                                    aria-hidden="true"
                                />
                            </a>
                        </div>
                    </div>
                </div>
            </div>
        </footer>
    );
};
