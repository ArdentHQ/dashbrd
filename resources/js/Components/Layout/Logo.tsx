import { Link } from "@inertiajs/react";
import cn from "classnames";
import { useTranslation } from "react-i18next";
import { Heading } from "@/Components/Heading";
import { Icon } from "@/Components/Icon";
import { appName } from "@/Utils/app";

interface Properties extends React.HTMLAttributes<HTMLDivElement> {
    renderDotSeparator?: boolean;
    displayHeading?: boolean;
    className?: string;
    hideSmallIcon?: boolean;
}

export const Logo = ({
    renderDotSeparator = false,
    displayHeading = false,
    className,
    hideSmallIcon = false,
}: Properties): JSX.Element => {
    const { t } = useTranslation();

    return (
        <div
            className={cn("flex items-center", className)}
            data-testid="Logo"
        >
            <Link
                href={t("urls.landing")}
                rel="noreferrer"
                className="flex items-center rounded-full outline-none outline-3 outline-offset-4 focus-visible:outline-theme-primary-300"
                data-testid="Logo__link"
            >
                <Icon
                    name="Logo"
                    aria-hidden="true"
                    size="2xl"
                    className={cn({
                        block: hideSmallIcon,
                        "hidden sm:block": !hideSmallIcon,
                    })}
                    data-testid="Logo__icon_medium"
                />

                <Icon
                    name="Logo"
                    aria-hidden="true"
                    size="xl"
                    className={cn({
                        "sm:hidden": !hideSmallIcon,
                        hidden: hideSmallIcon,
                    })}
                    data-testid="Logo__icon_small"
                />

                {displayHeading ? (
                    <Heading
                        level={3}
                        className={"hidden select-none sm:ml-2 sm:block"}
                    >
                        {appName()}
                    </Heading>
                ) : (
                    <span className="sr-only">{appName()}</span>
                )}
            </Link>

            {renderDotSeparator && <DotSeparator className="ml-4 mr-1.5" />}
        </div>
    );
};

const DotSeparator = ({ className, ...properties }: React.HTMLAttributes<HTMLDivElement>): JSX.Element => (
    <div
        aria-hidden="true"
        className={cn("h-1 w-1 rounded-full bg-theme-secondary-300 p-px sm:hidden", className)}
        data-testid="DotSeparator"
        {...properties}
    ></div>
);
