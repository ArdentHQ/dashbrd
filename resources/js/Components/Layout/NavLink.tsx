import { Link } from "@inertiajs/react";
import cn from "classnames";

interface Properties {
    href: string;
    active: boolean;
    children: React.ReactNode;
}

export const NavLink = ({ href, active, children }: Properties): JSX.Element => (
    <Link
        data-testid="NavLink"
        href={href}
        className={cn(
            "inline-flex items-center border-b-2 px-1 pt-1 text-sm font-medium leading-5 transition duration-150 ease-in-out focus:outline-none",
            {
                "border-indigo-400 focus:border-indigo-700 text-theme-secondary-900": active,
                "border-transparent text-theme-secondary-500 hover:border-theme-secondary-300 hover:text-theme-secondary-700 focus:border-theme-secondary-300 focus:text-theme-secondary-700":
                    !active,
            },
        )}
    >
        {children}
    </Link>
);
