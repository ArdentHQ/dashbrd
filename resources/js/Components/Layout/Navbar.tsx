import { Link } from "@inertiajs/react";
import cn from "classnames";
import { useTranslation } from "react-i18next";
import { Button } from "@/Components/Buttons/Button";
import { Icon } from "@/Components/Icon";
import { AppMenu } from "@/Components/Navbar/AppMenu";
import { MobileMenu } from "@/Components/Navbar/MobileMenu";
import { UserDetails } from "@/Components/Navbar/UserDetails";
import { type MetaMaskState } from "@/Hooks/useMetaMask";
import { isTruthy } from "@/Utils/is-truthy";

interface Properties
    extends React.HTMLAttributes<HTMLDivElement>,
        Pick<MetaMaskState, "connecting" | "initialized" | "connectWallet"> {
    authenticated: boolean;
    wallet: App.Data.AuthData["wallet"];
    user?: App.Data.UserData;
    isMaintenanceModeActive?: boolean;
}

export const Navbar = ({
    className,
    connecting,
    wallet,
    user,
    initialized,
    connectWallet,
    authenticated,
    isMaintenanceModeActive,
    ...properties
}: Properties): JSX.Element => {
    const { t } = useTranslation();

    const isAuthenticated = authenticated && isTruthy(wallet) && isTruthy(user);

    const renderAddress = (): JSX.Element => {
        if (isAuthenticated) {
            const { collectionCount, galleryCount } = wallet;

            return (
                <UserDetails
                    wallet={wallet}
                    collectionCount={collectionCount}
                    galleriesCount={galleryCount}
                    currency={user.attributes.currency}
                />
            );
        }

        return (
            <Button
                className="space-x-1"
                disabled={isTruthy(isMaintenanceModeActive) || connecting || !initialized}
                onClick={() => {
                    void connectWallet();
                }}
            >
                <span data-testid="Navbar__connect">{t("common.connect")}</span>

                <span className="hidden sm:inline">{t("common.wallet")}</span>
            </Button>
        );
    };

    return (
        <nav
            data-testid="Navbar"
            className={cn(
                "flex h-14 items-center justify-between border-b border-theme-secondary-300 bg-white px-6 py-2 xs:h-18 sm:px-8 sm:py-0",
                className,
            )}
            {...properties}
        >
            <div className="flex items-center">
                <MobileMenu
                    wallet={wallet}
                    connectWallet={connectWallet}
                    currency={user?.attributes.currency}
                    isConnectButtonDisabled={isTruthy(isMaintenanceModeActive) || connecting || !initialized}
                />
                <div className="flex items-center sm:space-x-4">
                    <Logo />

                    <AppMenu />
                </div>
            </div>

            <div className="hidden items-center sm:space-x-3 md-lg:flex">{renderAddress()}</div>
        </nav>
    );
};

const Logo = (): JSX.Element => (
    <div className="flex items-center">
        <Link
            href={route("dashboard")}
            target="_blank"
            rel="noreferrer"
            className="flex items-center rounded-full outline-none outline-3 outline-offset-4 focus-visible:outline-theme-hint-300"
        >
            <Icon
                name="Logo"
                size="2xl"
                className="hidden sm:block"
            />

            <Icon
                name="Logo"
                size="xl"
                className="sm:hidden"
            />
        </Link>
    </div>
);
