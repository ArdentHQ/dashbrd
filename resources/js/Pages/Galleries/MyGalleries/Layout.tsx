import { type ReactNode } from "react";
import { MyGalleryListboxMenu } from "./Components/MyGalleryListboxMenu";
import { MyGallerySidebar } from "./Components/MyGallerySidebar";
import LeftMenuLayout from "@/Layouts/LeftMenuLayout";

interface Properties {
    title: string;
    children: ReactNode;
    nftCount: number;
    galleryCount: number;
    draftsCount: number;
    isLoadingDrafts: boolean;
}

const Layout = ({ title, children, nftCount, galleryCount, draftsCount, isLoadingDrafts }: Properties): JSX.Element => (
    <LeftMenuLayout
        mobileMenu={
            <MyGalleryListboxMenu
                nftCount={nftCount}
                publishedCount={galleryCount}
                draftsCount={draftsCount}
            />
        }
        sidebarMenu={
            <MyGallerySidebar
                publishedCount={galleryCount}
                draftsCount={draftsCount}
                isLoadingDrafts={isLoadingDrafts}
            />
        }
        title={title}
    >
        {children}
    </LeftMenuLayout>
);
export default Layout;
