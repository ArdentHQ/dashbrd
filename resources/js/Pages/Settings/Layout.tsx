import { type ReactNode } from "react";
import { SettingsListboxMenu } from "./Components/SettingsListboxMenu";
import { SettingsSidebar } from "./Components/SettingsSidebar";
import { type ToastMessage } from "@/Components/Toast";
import LeftMenuLayout from "@/Layouts/LeftMenuLayout";

const Layout = ({
    title,
    toastMessage,
    children,
    mustBeSigned = false,
    showBackButton,
}: {
    title: string;
    children: ReactNode;
    toastMessage?: ToastMessage;
    mustBeSigned?: boolean;
    showBackButton?: boolean;
}): JSX.Element => (
    <LeftMenuLayout
        toastMessage={toastMessage}
        mustBeSigned={mustBeSigned}
        showBackButton={showBackButton}
        mobileMenu={<SettingsListboxMenu />}
        sidebarMenu={<SettingsSidebar />}
        title={title}
    >
        {children}
    </LeftMenuLayout>
);

export default Layout;
