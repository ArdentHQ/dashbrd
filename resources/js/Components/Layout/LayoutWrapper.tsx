import cn from "classnames";
import { type HTMLAttributes, useState } from "react";
import { AuthOverlay } from "@/Components/Layout/AuthOverlay";
import { Footer } from "@/Components/Layout/Footer";
import { Navbar } from "@/Components/Layout/Navbar";
import { SliderContext } from "@/Components/Slider";
import { ToastContainer, type ToastMessage } from "@/Components/Toast";
import { useAuth } from "@/Contexts/AuthContext";
import { useMetaMaskContext } from "@/Contexts/MetaMaskContext";
import { useAuthOverlay } from "@/Hooks/useAuthOverlay";

interface LayoutWrapperProperties extends HTMLAttributes<HTMLDivElement> {
    useVerticalOffset?: boolean;
    withSlider?: boolean;
    toastMessage?: ToastMessage;
    isMaintenanceModeActive?: boolean;
    belowHeader?: React.ReactNode;
    mustBeSigned?: boolean;
    displayAuthOverlay?: boolean;
    showBackButton?: boolean;
}

export const LayoutWrapper = ({
    children,
    belowHeader,
    className,
    useVerticalOffset = true,
    withSlider = false,
    toastMessage,
    isMaintenanceModeActive,
    mustBeSigned = false,
    displayAuthOverlay = true,
    showBackButton = false,
}: LayoutWrapperProperties): JSX.Element => {
    const { authenticated, signed, wallet, user, logout } = useAuth();

    const { showAuthOverlay, showCloseButton, closeOverlay } = useAuthOverlay({
        mustBeSigned,
    });

    const { connectWallet, initialized, connecting } = useMetaMaskContext();

    const content = (
        <div
            data-testid="LayoutWrapper"
            className="flex min-h-screen flex-col"
        >
            <header className="transition-default relative z-50 border-b border-theme-secondary-300 bg-white dark:border-theme-dark-700 dark:bg-theme-dark-900">
                <Navbar
                    connectWallet={connectWallet}
                    initialized={initialized}
                    connecting={connecting}
                    authenticated={authenticated}
                    wallet={wallet}
                    user={user ?? undefined}
                    isMaintenanceModeActive={isMaintenanceModeActive}
                    onLogout={() => {
                        void logout();
                    }}
                />
            </header>

            {displayAuthOverlay && (
                <AuthOverlay
                    show={showAuthOverlay}
                    showCloseButton={showCloseButton}
                    mustBeSigned={mustBeSigned}
                    closeOverlay={closeOverlay}
                    showBackButton={showBackButton}
                    // If we show the auth overlay even if the state says we are signed,
                    // means that the session has expired
                    sessionMayExpired={authenticated && signed}
                />
            )}

            {belowHeader}

            <div
                id="layout"
                className={cn("flex flex-1 flex-col")}
            >
                <main
                    className={cn(
                        "mx-auto flex w-full max-w-content flex-1 flex-col",
                        {
                            "py-6 sm:py-8": useVerticalOffset,
                        },
                        className,
                    )}
                >
                    {children}
                </main>

                <Footer withActionToolbar={withSlider} />
            </div>

            <ToastContainer toastMessage={toastMessage} />
        </div>
    );

    if (withSlider) {
        const [isSliderOpen, setSliderOpen] = useState(false);

        return (
            <SliderContext.Provider value={{ isOpen: isSliderOpen, setOpen: setSliderOpen }}>
                {content}
            </SliderContext.Provider>
        );
    }

    return content;
};
