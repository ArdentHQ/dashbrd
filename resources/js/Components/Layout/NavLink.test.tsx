import React from "react";
import { NavLink } from "@/Components/Layout/NavLink";
import { render, screen } from "@/Tests/testing-library";

describe("NavLink", () => {
    it("should render", () => {
        render(
            <NavLink
                href="/"
                active={false}
            >
                <span data-testid="test" />
            </NavLink>,
        );

        expect(screen.getByTestId("NavLink")).toBeInTheDocument();
        expect(screen.getByTestId("test")).toBeInTheDocument();
    });

    it("should render as active", () => {
        const { asFragment } = render(
            <NavLink
                href="/"
                active
            >
                <span data-testid="test" />
            </NavLink>,
        );

        expect(screen.getByTestId("NavLink")).toBeInTheDocument();
        expect(screen.getByTestId("test")).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();
    });
});
