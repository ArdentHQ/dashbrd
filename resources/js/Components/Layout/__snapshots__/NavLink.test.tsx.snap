// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavLink > should render as active 1`] = `
<DocumentFragment>
  <a
    class="inline-flex items-center border-b-2 px-1 pt-1 text-sm font-medium leading-5 transition duration-150 ease-in-out focus:outline-none border-indigo-400 focus:border-indigo-700 text-theme-secondary-900"
    data-testid="NavLink"
    href="/"
  >
    <span
      data-testid="test"
    />
  </a>
</DocumentFragment>
`;
